### PR TITLE
Include functionalities and updated structures

### PR DESCRIPTION
**Pull request by joaobrittoneto**

### CODE REVIEW

File: src/Driver.cpp [hunk 1]
```diff
@@ -275,8 +275,9 @@ string Driver::waitResponse(string const &command, CommandResponse expected)
         throw runtime_error("USBL Driver.cpp waitResponse: For the command: \""+ command +"\", device didn't send a response in " + to_string(time_out.toSeconds()) + " seconds time-out");
     // In DATA mode, validate response and return content without header.
     if(mode == DATA)
-    {
-        validResponse(response_info.buffer, command);
+    {   // Buffer validation of indicated length was displaced for extract packet.
+        // No need to do it here again.
+//        validResponse(response_info.buffer, command);
```
Comment (doudou) [inline src/Driver.cpp]: Then delete the line (but keep the comment !)

File: src/Driver.cpp [hunk 2]
```diff
@@ -336,8 +337,10 @@ Notification Driver::isNotification(string const &buffer)
     Notification notification = usblParser.findNotification(buffer);
     if(notification != NO_NOTIFICATION)
     {
-        if(mode == DATA)
-            validNotification(buffer);
+        // Buffer validation of indicated length was displaced for extract packet.
+        // No need to do it here again.
+//        if(mode == DATA)
```
Comment (doudou) [inline src/Driver.cpp]: Same here, keep the comment but delete the code

File: src/DriverTypes.hpp [hunk 3]
```diff
@@ -353,16 +353,22 @@ struct DeviceSettings
     // 8096.. 2097152
     std::vector<int> poolSize;
 
-    // Dropped data from transmission buffer (bytes) per channel.
-    // Set 0 for reset counter;
-    // Cases: 1)ResetType; 2)idleTimeout; 3)transmission to remoteAddress 0
-    std::vector<int> dropCount;
+//    // Dropped data from transmission buffer (bytes) per channel.
```
Comment (doudou) [inline src/DriverTypes.hpp]: If it's not in use, remove the code ! It's in git, you can always get it back later if you need it.

File: src/DriverTypes.hpp [hunk 4]
```diff
@@ -353,16 +353,22 @@ struct DeviceSettings
...
-    // Current Overflow count of the channel per channel.
-    // Set 0 for reset counter;
-    std::vector<int> overflowCounter;
+//    // Current Overflow count of the channel per channel.
+//    // Set 0 for reset counter;
+//    std::vector<int> overflowCounter;
 
+    // True: Reset dropCount
+    // False: Do not reset dropCount
+    bool resetDropCount;
```
Comment (doudou) [inline src/DriverTypes.hpp]: Not in use anywhere.

File: src/DriverTypes.hpp [hunk 5]
```diff
@@ -353,16 +353,22 @@ struct DeviceSettings
...
 
+    // True: Reset overflowCounter
+    // Flase: Do not reset overflowCounter
+    bool resetOverflowCounter;
```
Comment (doudou) [inline src/DriverTypes.hpp]: Not in use anywhere

File: src/DriverTypes.hpp (hunk 4 above)
Comment (joaobrittoneto) [inline src/DriverTypes.hpp]: It is used in the component, as a property. Should it be moved there?

File: src/DriverTypes.hpp (hunk 5 above)
Comment (joaobrittoneto) [inline src/DriverTypes.hpp]: It is used in the component, as a property. Should it be moved there?

**Comment by joaobrittoneto** [comment]
Unused code removed.

File: src/DriverTypes.hpp (hunk 5 above)
Comment (doudou) [inline src/DriverTypes.hpp]: If it is not in use in the library, it should not be in the library. So, yes.

**Pull request merged**